src/appkit.cs

```diff
@@ -27594,4 +28119,97 @@ interface NSPreviewRepresentingActivityItem : NSPreviewRepresentableActivityItem
 		NativeHandle Constructor (NSObject item, [NullAllowed] string title, [NullAllowed] NSItemProvider imageProvider, [NullAllowed] NSItemProvider iconProvider);
 	}
 
+	[NoMacCatalyst, Mac (14, 0)]
+	[BaseType (typeof (NSObject))]
+	interface NSNibConnector : NSCoding {
+		[NullAllowed, Export ("source", ArgumentSemantic.Weak)]
+		NSObject Source { get; set; }
+
+		[NullAllowed, Export ("destination", ArgumentSemantic.Weak)]
+		NSObject Destination { get; set; }
+
+		[Export ("label")]
+		string Label { get; set; }
+
+		[Export ("replaceObject:withObject:")]
+		void Replace (NSObject oldObject, NSObject newObject);
+
+		[Export ("establishConnection")]
+		void EstablishConnection ();
+	}
+
+	[NoMacCatalyst, Mac (14, 0)]
+	[BaseType (typeof (NSNibConnector))]
+	interface NSNibControlConnector {
+		[Export ("establishConnection")]
+		void EstablishConnection ();
+	}
+
+	[NoMacCatalyst, Mac (14, 0)]
+	[BaseType (typeof (NSView))]
+	interface NSTextInsertionIndicator {
+		[Export ("displayMode", ArgumentSemantic.Assign)]
+		NSTextInsertionIndicatorDisplayMode DisplayMode { get; set; }
+
+		[NullAllowed, Export ("color", ArgumentSemantic.Copy)]
+		NSColor Color { get; set; }
+
+		[Export ("automaticModeOptions", ArgumentSemantic.Assign)]
+		NSTextInsertionIndicatorAutomaticModeOptions AutomaticModeOptions { get; set; }
+
+		[NullAllowed, Export ("effectsViewInserter", ArgumentSemantic.Copy)]
+		Action<NSView> EffectsViewInserter { get; set; }
+	}
+
+	[NoMacCatalyst, Mac (14, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface NSMenuItemBadge : NSCopying {
+		[Static]
+		[Export ("updatesWithCount:")]
+		NSMenuItemBadge CreateUpdates (nint itemCount);
+
+		[Static]
+		[Export ("newItemsWithCount:")]
+		NSMenuItemBadge CreateNewItems (nint itemCount);
```

Comment:
Why do you think this method doesn't need the `[return: Release]` attribute?

Reply:
Because it is mean to be a lavel/menu that says "New Items" and does not mean it is a "new" method, is a factory method.

Reply:
That doesn't really matter, the semantics are built into the language. If the selector is named `new*`, Objective-C will return a retained value, no matter what the actual method does (unless it goes through some really weird hoops to not do what the language is supposed to do - and in that case, the method must be annotated with the corresponding Objective-C attribute to tell any callers that it doesn't return a retained value, and the method doesn't have such an attribute).

Reply:
Just to make sure that we did not make a mistake I wrote the following in objc in the viewDidLoad:
```objc
    NSMenuItem *newItemsItem = [[NSMenuItem alloc] initWithTitle:@"New Items" action:nil keyEquivalent:@""];
    newItemsItem.badge = [NSMenuItemBadge newItemsWithCount:3];
    [menu addItem:newItemsItem];
    
    NSLog(@"Retain count newItems is %ld", CFGetRetainCount((__bridge CFTypeRef)newItemsItem));

    // Add an alerts style badge.
    NSMenuItem *alertsItem = [[NSMenuItem alloc] initWithTitle:@"New Items" action:nil keyEquivalent:@""];
    alertsItem.badge = [NSMenuItemBadge alertsWithCount:4];
    [menu addItem:alertsItem];

    NSLog(@"Retain count alertItems is %ld", CFGetRetainCount((__bridge CFTypeRef)alertsItem));
    
    // Add an update style badge.
    NSMenuItem *updatesItem = [[NSMenuItem alloc] initWithTitle:@"Updates" action:nil keyEquivalent:@""];
    updatesItem.badge = [NSMenuItemBadge updatesWithCount:5];
    [menu addItem:updatesItem];
    
    NSLog(@"Retain count updateItems is %ld", CFGetRetainCount((__bridge CFTypeRef)updatesItem));
```
With the following output:
```
Retain count newItems is 2
Retain count alertItems is 2
Retain count updateItems is 2
```

Does it mean that all of the methods need the release then?

Reply:
You really need to look at the disassembly to figure out what happens.

The following Objective-C code:

```objective-c
void AAA ()
{
    [NSMenuItemBadge newItemsWithCount:3];
}

void BBB ()
{
    [NSMenuItemBadge alertsWithCount:4];
}

void CCC ()
{
    [NSMenuItemBadge updatesWithCount:5];
}
```

results in:

```
(lldb) disass -n AAA
macostest`AAA:
    0x10021f024 <+0>:  stp    x29, x30, [sp, #-0x10]!
    0x10021f028 <+4>:  mov    x29, sp
    0x10021f02c <+8>:  adrp   x8, 5
    0x10021f030 <+12>: ldr    x0, [x8, #0xa58]
    0x10021f034 <+16>: mov    x2, #0x3
    0x10021f038 <+20>: bl     0x10021f2f8               ; objc_msgSend$newItemsWithCount:
    0x10021f03c <+24>: bl     0x10021f290               ; symbol stub for: objc_release
    0x10021f040 <+28>: ldp    x29, x30, [sp], #0x10
    0x10021f044 <+32>: ret    

(lldb) disass -n BBB
macostest`BBB:
    0x10021f048 <+0>:  stp    x29, x30, [sp, #-0x10]!
    0x10021f04c <+4>:  mov    x29, sp
    0x10021f050 <+8>:  adrp   x8, 5
    0x10021f054 <+12>: ldr    x0, [x8, #0xa58]
    0x10021f058 <+16>: mov    x2, #0x4
    0x10021f05c <+20>: bl     0x10021f2d8               ; objc_msgSend$alertsWithCount:
    0x10021f060 <+24>: mov    x29, x29
    0x10021f064 <+28>: ldp    x29, x30, [sp], #0x10
    0x10021f068 <+32>: b      0x10021f2c0               ; symbol stub for: objc_unsafeClaimAutoreleasedReturnValue

(lldb) disass -n CCC
macostest`CCC:
    0x10021f06c <+0>:  stp    x29, x30, [sp, #-0x10]!
    0x10021f070 <+4>:  mov    x29, sp
    0x10021f074 <+8>:  adrp   x8, 5
    0x10021f078 <+12>: ldr    x0, [x8, #0xa58]
    0x10021f07c <+16>: mov    x2, #0x5
    0x10021f080 <+20>: bl     0x10021f318               ; objc_msgSend$updatesWithCount:
    0x10021f084 <+24>: mov    x29, x29
    0x10021f088 <+28>: ldp    x29, x30, [sp], #0x10
    0x10021f08c <+32>: b      0x10021f2c0               ; symbol stub for: objc_unsafeClaimAutoreleasedReturnValue
```

Note how `objc_release` is called for `newItemsWithCount:`, but not the other two selectors (but then `objc_unsafeClaimAutoreleasedReturnValue` is called for the other two selectors, but not `newItemsWithCount:`)